### DISCUSSION TITLE
Slampher R2 V2.4 Program Mode

### DISCUSSION THREAD

**Discussion by Westiewombat** [arendst/Tasmota#13044] [Q&A] [answered]
I have connected up 3.3v, RX, TX, Gnd using headers - Green LED Flashes ok

I cant seem to get the unit into program mode so I can Tasmotizer.

Any suggestions on the sequence - I have held down the push button until flashes red, but no luck

Thanks for any assistance

**Comment by barbudor** [comment, marked as answer]
You should  held the button, then power up the device
Pressing the button after the device has boot is too late to enter flashing mode

**Reply by barbudor** [reply]
From https://blog.hobbytronics.pk/tasmota-sonoff-slampher-local-control/

> The Sonoff Slampher have an 8bit EFM8BB10F2G-A-QFN20 microcontroller that listens to the radio module messages and handles the on-board button tied to GPIO-0. However, it can not be used to enter Sonoff Slampher into programming mode. For starting Sonoff Slampher in programming mode GPIO-0 should be tied to ground. For this purpose, solder a small piece of wire to ESP8285 side of resistor R9 which is directly connected to GPIO-0.

So pressing button will be useless.
You need to make the connection as reported

You can see the above site from Google cache : https://webcache.googleusercontent.com/search?q=cache:MpQtwaxkSBIJ:https://hobbytronics.com.pk/tasmota-sonoff-slampher-local-control/+&cd=1&hl=fr&ct=clnk&gl=fr

**Reply by Westiewombat** [reply]
Thank you @barbudor . The holding down the button whilst powering up did the trick. I was able then to Tasmonize successfully.  I didn't have to solder the GPIO0 to ground - perhaps I was lucky with my timing or v2.4 didn't require it... anyway I have a few more Slamphers to Tasmonize so I'll report back any problems and if I have issues will try above

NB My browser was show some  issues access the hobbytronics blog web site, saying the site had an error 526 Certificate Error on Cloudflare, so thanks for copying out of cache
